lib/smb.c

```diff
@@ -688,7 +688,7 @@ static CURLcode smb_connection_state(struct connectdata *conn, bool *done)
       return CURLE_COULDNT_CONNECT;
     }
     nrsp = msg;
-    memcpy(smbc->challenge, nrsp->bytes, sizeof(smbc->challenge));
+    memcpy(smbc->challenge, nrsp->bytes, sizeof(nrsp->bytes));
```

Comment:
isn't this covered above that we know there will be bytes >= smbc->challenge, and here we only want those bytes so why are we going to copy all the bytes in the response to the challenge? this is what @bagder is talking about